etl/steps/__init__.py

```diff
@@ -109,6 +113,32 @@ def reverse_graph(graph: Graph) -> Graph:
     return dict(g)
 
 
+def maximal_subgraph(graph: Graph, nodes: Set[str]) -> Graph:
+    """
+    Take a graph and a set of nodes and return a subgraph that only includes
+    those nodes and their dependencies. (Return subgraph that contains all nodes
+    reachable from given nodes... is there a name for it?)
+    """
+    subgraph = defaultdict(set)
+
+    while nodes:
+        node = nodes.pop()
+        # already traversed
+        if node in subgraph:
+            continue
+        subgraph[node] = set(graph.get(node, set()))
+        nodes |= subgraph[node]
+
+    return dict(subgraph)
+
+
+def graph_nodes(graph: Graph) -> Set[str]:
+    all_steps = set(graph)
+    for children in graph.values():
+        all_steps.update(children)
+    return all_steps
```

Comment:
Nice fix!

It looks like `graph_nodes()` is never used, you can kill it.

Reply:
My bad, gonna fix that in future PR.